### PR TITLE
(docs) make demo page more keyboard accessible

### PR DESCRIPTION
**Pull request by Hirse**
### Changes
Adds basic keyboard accessibility to the [demo page](https://highlightjs.org/static/demo) by adding `a`-tags to react to keydown events.

hljs-demo-keyboard

### Checklist
- [ ] ~Added markup tests~, or they don't apply here because no highlight changes
- [ ] ~Updated the changelog at `CHANGES.md`~ No published changes

### CODE REVIEW

**Comment by joshgoebel** [comment]
> The onclick event handler (and click event) is triggered when the mouse is pressed and released when over a page element or when the Enter key is pressed while a keyboard-navigable element has focus. 

Shouldn't the existing `onclick` handler just work?  A target of "#" could be added if necessary...

**Comment by Hirse** [comment]
Added `<a>`-tags and removed custom keydown handler.

File: demo/demo.js [hunk 1]
```diff
@@ -1,20 +1,13 @@
...
+  category.addEventListener("click", (event) => {
+    event.preventDefault();
 
-document.querySelectorAll(".categories > li").forEach((category) => {
-  addEventListeners(category, (event) => {
     const current = document.querySelector(".categories .current");
-    const currentCategory = current.dataset.category;
-    const nextCategory = event.target.dataset.category;
+    const currentCategory = current.name;
+    const nextCategory = event.target.name;
```
Comment (joshgoebel) [inline demo/demo.js:10]: Using dataset seems entirely appropriate here to name the type of data vs using the generic HTML `name`... was there a reason for this change?

**Comment by joshgoebel** [comment]
I really like that all the tabindex=0 disappears, feels like a step in the correct direction.

File: demo/demo.js [hunk 2]
```diff
@@ -23,15 +25,21 @@ document.querySelectorAll(".categories > li").forEach((category) => {
   });
 });
 
-document.querySelectorAll(".styles > li").forEach((style) => {
+document.querySelectorAll(".styles a").forEach((style) => {
```
Comment (joshgoebel) [inline demo/demo.js:28]:
```suggestion
document.querySelectorAll(".styles li a").forEach((style) => {
```

More descriptive as these are still a list.

Comment (joshgoebel) [reply demo/demo.js:28]: Or perhaps just `ul.styles` ?

File: demo/demo.js (hunk 1 above)
Comment (Hirse) [reply demo/demo.js:10]: No, just copying from my testing. :D

File: demo/demo.js (hunk 2 above)
Comment (Hirse) [reply demo/demo.js]: Changed to your first suggestion.

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**